### ISSUE TITLE
Portal for GPG encryption/decryption

### ISSUE THREAD

**Issue by AdrianKoshka** [flatpak/xdg-desktop-portal#178]
Heyo, I recently made Thunderbird available as a [fatpak on flathub](https://github.com/flathub/org.mozilla.Thunderbird), but right now Enigmail/GnuPG functionality doesn't work. I don't want to expose the entire gpg directory, keys and all, to applications for it though if possible.

**Comment by A6GibKm** [comment]
Same issue while trying to do a variety of tasks in emacs, signing commits, read a password from a encrypted file, etc.

**Comment by TingPing** [comment]
To make progress here I'd say an application that relies on this needs to write out its API needs (and ideally make an implementation).

**Comment by muelli** [comment]
seems nobody has suggested to use the [Seahorse API](https://gitlab.gnome.org/GNOME/seahorse/-/blob/SEAHORSE_0_9_10/tests/dbus-crypto-test.py) just yet, but it may be worth a shot.

**Comment by RomanRiabenko** [comment]
`--socket=gpg-agent` was added with  [flatpak/flatpak#4958](https://github.com/flatpak/flatpak/pull/4958) in version 1.14.0.

**Comment by TingPing** [comment]
I don't believe that invalidates this issue. `gpg-socket` is a sandbox bypass and not a portal.

**Comment by nanonyme** [comment]
@TingPing does the agent leak secret material to sandbox though? If not, the difference isn't that big.
